cloudify_agent/api/plugins/installer.py

```diff
@@ -350,81 +351,83 @@ def _rmtree(path):
-    # of this logger - by name, before importing pip. (see CFY-4866)
-    _previous_signal = []
-    _previous_level = []
-
-    def _patch_pip_download():
-        pip_utils_logger = logging.getLogger('pip.utils')
-        _previous_level.append(pip_utils_logger.level)
-        pip_utils_logger.setLevel(logging.CRITICAL)
-
-        try:
-            import pip.utils.ui
-
-            def _stub_signal(sig, action):
-                return None
-            if hasattr(pip.utils.ui, 'signal'):
-                _previous_signal.append(pip.utils.ui.signal)
-                pip.utils.ui.signal = _stub_signal
-        except ImportError:
-            pass
-
-    def _restore_pip_download():
-        try:
-            import pip.utils.ui
-            if hasattr(pip.utils.ui, 'signal') and _previous_signal:
-                pip.utils.ui.signal = _previous_signal[0]
-        except ImportError:
-            pass
-        pip_utils_logger = logging.getLogger('pip.utils')
-        pip_utils_logger.setLevel(_previous_level[0])
 
+    """
     plugin_dir = None
+    archive_dir = tempfile.mkdtemp()
+    runner = LocalCommandRunner()
+
     try:
-        plugin_dir = tempfile.mkdtemp()
-        _patch_pip_download()
-        # Import here, after patch
-        import pip
-        pip.download.unpack_url(link=pip.index.Link(package_url),
-                                location=plugin_dir,
-                                download_dir=None,
-                                only_download=False)
-    except Exception as e:
+        command = [get_pip_path(), 'download', '-d',
+                   archive_dir, '--no-deps', package_url]
+        runner.run(command=command)
+        archive = _get_archive(archive_dir, package_url)
+        plugin_dir_parent = extract_archive(archive)
+        plugin_dir = _get_plugin_path(plugin_dir_parent, package_url)
+
+    except NonRecoverableError as e:
         if plugin_dir and os.path.exists(plugin_dir):
             shutil.rmtree(plugin_dir)
-        raise exceptions.PluginInstallationError(
-            'Failed to download and unpack package from {0}: {1}.'
-            'You may consider uploading the plugin\'s Wagon archive '
-            'to the manager, For more information please refer to '
-            'the documentation.'.format(package_url, str(e)))
+        raise e
+
     finally:
-        _restore_pip_download()
+        if os.path.exists(archive_dir):
+            shutil.rmtree(archive_dir)
 
     return plugin_dir
 
 
+def _get_plugin_path(plugin_dir_parent, package_url):
+    """
+    plugin_dir_parent is a directory containing the plugin dir. Since we
+    need the plugin`s dir, we find it's name and concatenate it to the
+    plugin_dir_parent.
+    """
+    contents = list(walk(plugin_dir_parent))
+    if len(contents) < 1:
+        _remove_tempdir_and_raise_proper_exception(package_url,
+                                                   plugin_dir_parent)
+    plugin_dir_name = contents[0][1][0]
```

Comment:
It's succinct but not very readable. Let's maybe do it in two lines but make it more easy to understand.